test/prepare_disk.sh

```diff
@@ -0,0 +1,431 @@
+#!/bin/sh
+#
+# Copyright (C) 2018 Ultimaker B.V.
+# Copyright (C) 2018 Raymond Siudak <raysiudak@gmail.com>
+# Copyright (C) 2018 Olliver Schinagl <oliver@schinagl.nl>
+#
+# SPDX-License-Identifier: AGPL-3.0+
+
+set -eu
+
+ARM_EMU_BIN="${ARM_EMU_BIN:-}"
+
+CWD="$(pwd)"
+
+SYSTEM_UPDATE_DIR="${SYSTEM_UPDATE_DIR:-/etc/system_update}"
+PARTITION_TABLE_FILE="test_jedi_emmc_sfdisk.table"
+TARGET_STORAGE_DEVICE=""
+
+JEDI_PARTITION_TABLE_FILE_NAME="config/jedi_emmc_sfdisk.table"
+PREPARE_DISK_COMMAND="/etc/system_update.d/30_prepare_disk.sh"
+
+STORAGE_DEVICE_IMG="storage_device.img"
+MIN_PARTITION_SIZE="78124"    # 40 MiB (enough for f2fs and ext4)
+BYTES_PER_SECTOR="512"
+STORAGE_DEVICE_SIZE="7553024" # sectors, about 3.6 GiB
+BOOT_START="2048"             # offset 1 MiB
+ROOTFS_START="67584"          # offset 33 MiB
+USERDATA_START="1998848"      # offset 976 MiB
+
+TEST_OUTPUT_FILE="$(mktemp -d)/test_results_$(basename "${0%.sh}").txt"
+
+toolbox_image=""
+toolbox_root_dir=""
+work_dir=""
+
+exit_on_failure=false
+result=0
+
+# shellcheck source=test/include/armhf_emulation.sh
+. "test/include/armhf_emulation.sh"
+
+
+execute_prepare_disk()
+{
+    chroot_environment="SYSTEM_UPDATE_DIR=${SYSTEM_UPDATE_DIR}"
+    chroot_environment="${chroot_environment} PARTITION_TABLE_FILE=${PARTITION_TABLE_FILE}"
+    chroot_environment="${chroot_environment} TARGET_STORAGE_DEVICE=${TARGET_STORAGE_DEVICE}"
+
+    sha512sum "${PARTITION_TABLE_FILE}" > "${PARTITION_TABLE_FILE}.sha512"
+    cp "${PARTITION_TABLE_FILE}" "${toolbox_root_dir}${SYSTEM_UPDATE_DIR}/${PARTITION_TABLE_FILE}"
+    cp "${PARTITION_TABLE_FILE}.sha512" "${toolbox_root_dir}${SYSTEM_UPDATE_DIR}/${PARTITION_TABLE_FILE}.sha512"
+
+    chroot "${toolbox_root_dir}" /bin/sh -c "${chroot_environment} ${PREPARE_DISK_COMMAND}" || return 1
+
+    sfdisk -d "${TARGET_STORAGE_DEVICE}" > "${PARTITION_TABLE_FILE}.verify"
+
+    # Remove the identifiers in the header because they will always change.
+    sed -i "s/label-id:.*//" "${PARTITION_TABLE_FILE}"
+    sed -i "s/label-id:.*//" "${PARTITION_TABLE_FILE}.verify"
+
+    # Remove the name from the source partition file, as the names are not stored in the partition table.
+    sed -i "s/, name=.*//" "${PARTITION_TABLE_FILE}"
+
+    diff -b "${PARTITION_TABLE_FILE}" "${PARTITION_TABLE_FILE}.verify" || return 1
+}
+
+test_disk_integrity()
+{
+    set -e
```

Comment:
hmm, this looks like a change, and why is that needed on a per test basis? Either the set -eu (what happened to the u) is applied whenever the file is sourced, or from the sourcing file, i'd be supprised that you have to set it each call to the function. (Unless you are doing if test_function; then) then, i understand why you are doing this, but then we should really re-evaluate our testing framework.

Reply:
Yes this is wrong. The set statement is now only in the top of the file before everything else, all functions expected to exit on error are not executed in an 'if' statement.